plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -127,6 +128,10 @@ def set_replica_target
       template[:spec][:replicas] = @doc.replica_target if template[:kind] == 'Deployment'
     end
 
+    def set_name
+      template[:metadata][:name] = @doc.kubernetes_role.resource_name
```

Comment:
And I'm kind of lost of why this is needed. It doesn't seem nice if we change what the user already specified.

If ```deployment.spec.template.metadata.name``` is not specified (is not mandatory, in fact I don't have it in any of my projects :)), kubernetes will use something for us. And if the user specified it, I guess he/she might have a reason.

Why do we want to set this to something in particular? Sorry, I don't understand :-/